### PR TITLE
Cpp ut ci flow

### PR DESCRIPTION
**Pull request by hongdami**
Issue Number: 

Objective of pull request:

## Pull request checklist

Your PR fulfills the following requirements:
- [ ] [Issue](https://github.com/lava-nc/lava/issues) created that explains the change and why it's needed
- [ ] Tests are part of the PR (for bug fixes / features)
- [ ] [Docs](https://github.com/lava-nc/docs) reviewed and added / updated if needed (for bug fixes / features)
- [ ] PR conforms to [Coding Conventions](https://lava-nc.org/developer_guide.html#coding-conventions)
- [ ] [PR applys BSD 3-clause or LGPL2.1+ Licenses](https://lava-nc.org/developer_guide.html#add-a-license) to all code files
- [ ] Lint (`flakeheaven lint src/lava tests/`) and (`bandit -r src/lava/.`) pass locally
- [ ] Build tests (`pytest`) passes locally

## Pull request type

 

 

Please check your PR type:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -168,6 +168,9 @@ jobs:
       with:
         lfs: true
 
+    - name: Set msg lib build type
+      run: echo "DEBUG=0" >> $GITHUB_ENV
```
Comment (killight98) [inline .github/workflows/ci-build.yml:172]: refer to [this link](https://docs.github.com/en/actions/learn-github-actions/environment-variables)
```yaml
name: Greeting on variable day

on:
  workflow_dispatch

env:
  DAY_OF_WEEK: Monday

jobs:
  greeting_job:
    runs-on: ubuntu-latest
    env:
      Greeting: Hello
    steps:
      - name: "Say Hello Mona it's Monday"
        run: echo "$Greeting $First_Name. Today is $DAY_OF_WEEK!"
        env:
          First_Name: Mona
```

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -196,6 +199,11 @@ jobs:
       working-directory: ./src/lava/magma/runtime/message_infrastructure
       run: poetry install --no-interaction
 
+    - name: Run message infrastructure cpp unit tests
+      working-directory: ./src/lava/magma/runtime/message_infrastructure/
+      if: env.DEBUG == '1'
```
Comment (killight98) [inline .github/workflows/ci-build.yml:204]: remove this condition checking

File: src/lava/magma/runtime/message_infrastructure/docs/run_example.md [hunk 3]
```diff
@@ -11,3 +11,11 @@ cd ..
 source setenv.sh
 python test/test_example.py
 ```
+
+## Running Unit Tests (C++ / GoogleTest)
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/docs/run_example.md:15]: add a cmd for debug

File: src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+FetchContent_Declare(
+  googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG release-1.12.1
+)
+
+# For Windows: Prevent overriding the parent project's compiler/linker settings
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
+
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt:11]: create a sub-directory under __test__ for cpp unit tests as following:
```
message_infrastructure
| - test \
      | - cpptest\*.cc
```

Comment (seowwj) [reply src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt:11]: Resolved in commit [6cea024](https://github.com/hongdami/lava/pull/5/commits/6cea02411f6255e0ed30146afca7bbe9163224f3)

Comment (killight98) [reply src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt:11]: @seowwj pls push the commit under this PR.

**Comment by killight98** [comment]
@seowwj 
pls merge the branch [lava-nc:messaging_refactor_release2nd](https://github.com/lava-nc/lava/tree/messaging_refactor_release2nd) with new structure of msg infr lib.
need to do following steps:
```
- update test/CMakeLists
- update include path at uts
- do not use pybind11 & numpy & PY_WRAPPER_SRCS at cpp ut
```

**Comment by seowwj** [comment]
> @seowwj pls merge the branch [lava-nc:messaging_refactor_release2nd](https://github.com/lava-nc/lava/tree/messaging_refactor_release2nd) with new structure of msg infr lib. need to do following steps:
> 
> ```
> - update test/CMakeLists
> - update include path at uts
> - do not use pybind11 & numpy & PY_WRAPPER_SRCS at cpp ut
> ```

Resolved 👍

File: src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+add_executable(
+  test_messaging_infrastructure
+  cpptest/test_multiprocessing.cc
+  # cpptest/test_channel.cc
+)
+
+target_link_libraries(
+  test_messaging_infrastructure
+  GTest::gtest_main
+  pybind11::embed
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt:23]: remove this dependence

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_multiprocessing.cc [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    int return_value = mp.BuildActor(bound_fn);
+    std::cout << "Return Value --> " << return_value << std::endl;
+  }
+
+  sleep(1);
+
+  std::vector<AbstractActor::ActorPtr>& actorList = mp.GetActors();
+  std::cout << "Actor List Length --> " << actorList.size() << std::endl;
+  for (auto actor : actorList){
+    actor->ForceStop();
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/cpptest/test_multiprocessing.cc:89]: replace by mp.stop

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_multiprocessing.cc [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+// See: https://spdx.org/licenses/
+
+#include <iostream>
+
+#include <gtest/gtest.h>
+
+#include <message_infrastructure/csrc/core/multiprocessing.h>
+#include <message_infrastructure/csrc/core/abstract_actor.h>
+
+TEST(HelloTest, BasicAssertions) {
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/cpptest/test_multiprocessing.cc:12]: remove this example case.

**Review by killight98 — CHANGES_REQUESTED** [review]

**Review by killight98 — APPROVED** [review]

**Pull request merged**